src/dune/dune_file.ml

```diff
@@ -1765,20 +1770,29 @@ end
 
 module Deprecated_library_name = struct
   module Old_public_name = struct
+    type kind =
+      | Not_deprecated
+      | Deprecated of { deprecated_package : Package.Name.t }
+
     type t =
-      { deprecated : bool
+      { kind : kind
       ; public : Public_lib.t
       }
 
     let decode =
-      let+ public = Public_lib.decode ~allow_deprecated_names:true () in
-      let deprecated =
-        not
-          (Package.Name.equal
-             (Lib_name.package_name (Public_lib.name public))
-             (Public_lib.package public).name)
+      let+ public = Public_lib.decode ~allow_deprecated_names:true in
```

Comment:
I got lost in the logic below. Could you add a comment explaining the computation of `kind`? (I guess I'm missing the larger picture here: how deprecated packages work, so a couple of lines to explain what `kind` is would be helpful too.)

Reply:
Yeah, honestly it was confusing to me as well. Which is why I decided to start simplifying this code. I added a comment for this argument.